contracts/src/lib/CrossMsgHelper.sol

```diff
@@ -71,10 +71,11 @@ library CrossMsgHelper {
     /// and identifies the receipt through the hash of the original message.
     function createReceiptMsg(
         IpcEnvelope calldata crossMsg,
+        ReceiptType receiptType,
         bool success,
         bytes memory ret
     ) public pure returns (IpcEnvelope memory) {
-        ReceiptMsg memory message = ReceiptMsg({id: toHash(crossMsg), success: success, ret: ret});
+        ReceiptMsg memory message = ReceiptMsg({receiptType: receiptType, id: toHash(crossMsg), success: success, ret: ret});
```

Comment:
@cryptoAtwill This is a bit entropic, let's simplify:

- remove `receiptType`
- rename `success` to `outcome`
- add an `outcomeType` enum with three values: `Ok`, `ActorErr`, `SystemErr`.